File: Roman to Integer/Roman_to_Integer.swift
import UIKit

class Solution {
    func romanToInt(_ s: String) -> Int {
        let romanToIntDics: [Character:Int] = ["I":1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000]
        var changeS = s.replacingOccurrences(of: "IV", with: "IIII")
            .replacingOccurrences(of: "IX", with: "VIIII")
            .replacingOccurrences(of: "XL", with: "XXXX")
            .replacingOccurrences(of: "XC", with: "LXXXX")
            .replacingOccurrences(of: "CD", with: "CCCC")
            .replacingOccurrences(of: "CM", with: "DCCCC")
        var sum = 0
        
        var sum = changeS.map({ (ch: Character) -> Int in return romanToIntDics[ch]! }).reduce(0, +)
            sum += romanToIntDics[ch]!
        }

        return sum
    }
}

let solution = Solution()
print(solution.romanToInt("MCMXCIV"))

File: Roman to Integer/Roman to Integer.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>
#include <set>

using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        map<char, int> m{
            { 'I', 1 },
            { 'V', 5 },
            { 'X', 10 },
            { 'L', 50 },
            { 'C', 100 },
            { 'D', 500 },
            { 'M', 1000 }};

        if (s.size() == 1)
            return m[s.at(0)];

        int buf = m[s.at(s.size() - 1)];
        for (int i = s.size() -1; i > 0; i--)
        {
            if (m[s.at(i - 1)] < m[s.at(i)])
                buf -= m[s.at(i - 1)];
            else
                buf += m[s.at(i - 1)];
        }
        return buf;
    }
};

int main()
{
    Solution s;
    bool b = s.romanToInt("MCMXCIV");
    return 0;
}